File: code-generator/src/main/java/io/github/potjerodekool/openapi/internal/di/conditiontest/ConditionalOnMissingBeanTest.java
package io.github.potjerodekool.openapi.internal.di.conditiontest;

import io.github.potjerodekool.openapi.common.dependency.ApplicationContext;
import io.github.potjerodekool.openapi.internal.di.bean.BeanDefinition;
import io.github.potjerodekool.openapi.common.dependency.ConditionalOnMissingBean;

public class ConditionalOnMissingBeanTest implements ConditionalTest<ConditionalOnMissingBean> {

    private final ApplicationContext applicationContext;

    public ConditionalOnMissingBeanTest(final ApplicationContext applicationContext) {
        this.applicationContext = applicationContext;
    }

    @Override
    public boolean test(final ConditionalOnMissingBean condition,
                        final BeanDefinition beanDefinition) {
        return !applicationContext.isBeanOfTypePresent(condition.type());
    }
}

File: code-generator/src/main/kotlin/io/github/potjerodekool/openapi/internal/di/conditiontest/ConditionalOnMissingBeanTest.kt
package io.github.potjerodekool.openapi.internal.di.conditiontest

import io.github.potjerodekool.openapi.common.dependency.ApplicationContext
import io.github.potjerodekool.openapi.common.dependency.ConditionalOnMissingBean
import io.github.potjerodekool.openapi.internal.di.bean.BeanDefinition

class ConditionalOnMissingBeanTest(private val applicationContext: ApplicationContext) :
    ConditionalTest<ConditionalOnMissingBean> {
    override fun test(
        condition: ConditionalOnMissingBean,
        beanDefinition: BeanDefinition
    ): Boolean {
        return !applicationContext.isBeanOfTypePresent(condition.type)
    }
}
